src/slice.rs

```diff
@@ -70,7 +70,7 @@ impl Slice {
 /// Struct to hold all the header payload of a [`Slice`].
 ///
 /// This is included in each [`Shred`] after shredding.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
Do we need these here and below?  I like to introduce these only as needed because they just slow down compile times.